Sources/Core/CompilerInfo.swift

```diff
@@ -0,0 +1,69 @@
+/// Describes the compiler; used as source of truth for conditional compilation.
+public struct CompilerInfo {
+
+  /// A version number with multiple parts; form: x(.y)*
+  public struct VersionNumber: Codable, Comparable {
+
+    public let parts: [Int]
```

Comment:
```suggestion
    public let parts: (major: Int, minor: Int, path: Int)
```
What do you think about adopting semantic versioning?